dash-renderer/src/actions/dependencies_ts.ts

```diff
@@ -0,0 +1,323 @@
+import {
+    all,
+    assoc,
+    concat,
+    difference,
+    filter,
+    flatten,
+    forEach,
+    isEmpty,
+    keys,
+    map,
+    mergeWith,
+    partition,
+    pickBy,
+    props,
+    reduce,
+    zipObj
+} from 'ramda';
+import { ICallback, ICallbackProperty, ICallbackDefinition, ILayoutCallbackProperty, ICallbackTemplate } from '../types/callbacks';
+import { addAllResolvedFromOutputs, splitIdAndProp, stringifyId, getCallbacksInLayout, isMultiValued, idMatch } from './dependencies';
```

Comment:
Is there a pattern to what got moved to `.ts` and what's still in the `.js` file? Or is it just the things you modified you moved to ts?

Reply:
in particular `getLayoutCallbacks` vs `getCallbacksInLayout` is a bit confusing...

Reply:
It's a mixed bag. Some things are new, some things got moved to TS just to get a more precise function signature as it was used in TS and needed to be casted to `any`, etc.

Reply:
> getLayoutCallbacks vs getCallbacksInLayout 

Agreed

Reply:
https://github.com/plotly/dash/pull/1254/commits/484a4e5d8a58d2cd661be80e1158b2e1698fccb2